### PR TITLE
Makes sure xmlobj.CommonPrefixes is an array

### LINKED ISSUE #876: ERROR: xmlobj.CommonPrefixes.forEach is not a function
**minio version：^7.0.17**

</br>

 found this problem when I used the **listobjects** API

1、When the third parameter recursive of listobjects is set to true, the result returned is correct
```javascript
// code
const bucketName = params.bucketName || '';
const prefix = params.prefix || '';
const stream = app.minio.listObjects(bucketName, prefix, true);
return new Promise(resolve => {
    const res = [];
    stream.on('data', obj => {
        res.push(obj);
    });
    stream.on('end', () => {
        resolve({
            code: 0,
            data: res,
        });
    });
    stream.on('error', err => {
        resolve({
            code: 1,
            msg: err,
        });
    });
});

// result
[
    {
        "name":"child/aa.txt",
        "lastModified":"2020-12-01T04:04:18.148Z",
        "etag":"166bcae2d953f92b1fa3b40d218d2a38",
        "size":57
    },
    {
        "name":"bb.jpg",
        "lastModified":"2020-12-01T02:39:41.026Z",
        "etag":"ba02b3771a3ceeaf1d4925c62d05780b",
        "size":8730
    },
    {
        "name":"cc.jpg",
        "lastModified":"2020-12-01T02:40:21.905Z",
        "etag":"aed69181e64900aeaaf4b18798c7bcc4",
        "size":10891
    }
]
```

2、However, when the third parameter recursive is set to false, an error is reported

```javascript
// code
app.minio.listObjects(bucketName, prefix, false);
......

// result
ERROR: xmlobj.CommonPrefixes.forEach is not a function
```

I entered the wrong node_ modules\minio\dist\main\xml- parsers.js:381 : 27 files, printing xmlobj.CommonPrefixes The results are as follows：
```javascript
// Error source code
if (xmlobj.CommonPrefixes) {
    console.log('xmlobj.CommonPrefixes', xmlobj.CommonPrefixes);
    xmlobj.CommonPrefixes.forEach(function (commonPrefix) {
        var prefix = commonPrefix.Prefix[0];
        var size = 0;
        result.objects.push({
            prefix,
            size
        });
    });
}

// console.log result
xmlobj.CommonPrefixes { Prefix: 'child/' }
```

### PR DESCRIPTION
**Pull request by ebozduman**
Fixes #876 

When there is only one prefix in `xmlobj.CommonPrefix`, `parseXml` parses and returns a single item.
However, we always treat the object as an array. So, the fix makes sure `xmlobj.CommonPrefix` is always an array, by modifying a single string item to an array with a single element.

### CODE REVIEW

**Comment by ebozduman** [comment]
Fixes #876, #879, #880, and #881

File: src/main/xml-parsers.js [hunk 1]
```diff
@@ -268,10 +276,7 @@ export function parseListObjects(xml) {
   xmlobj = xmlobj.ListBucketResult
   if (xmlobj.IsTruncated && xmlobj.IsTruncated === 'true') xmlobj.isTruncated = true
```
Comment (kanagarajkm) [inline src/main/xml-parsers.js:277]: @ebozduman `xmlobj.IsTruncated` is returned as boolean not as a string by the parser. so we could do
```
if (xmlobj.IsTruncated && (xmlobj.IsTruncated === true || xmlobj.IsTruncated === 'true'))
```
This is what leads to issue #881

Comment (kanagarajkm) [reply src/main/xml-parsers.js:277]: `xmlobj.NextContinuationToken[0]` needs to be fixed as well

Comment (ebozduman) [reply src/main/xml-parsers.js:277]: Sorry @kanagarajkm 
This part has already changed.

Comment (ebozduman) [reply src/main/xml-parsers.js]: @kanagarajkm 
PTAL again. I also addressed issue #881.

**Comment by kanagarajkm** [comment]
@ebozduman `setBucketNotification` is throwing following error

```
      var builder = new _fastXmlParser.default.Builder({
                    ^

TypeError: _fastXmlParser.default.Builder is not a constructor
    at Client.setBucketNotification (/home/kanagaraj/workspace/minio-js/dist/main/minio.js:2568:21)
    at Client.<anonymous> (/home/kanagaraj/workspace/minio-js/dist/main/helpers.js:69:51)
    at Object.<anonymous> (/home/kanagaraj/workspace/minio-js/test.js:35:13)
    at Module._compile (internal/modules/cjs/loader.js:1158:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1178:10)
    at Module.load (internal/modules/cjs/loader.js:1002:32)
    at Function.Module._load (internal/modules/cjs/loader.js:901:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
    at internal/main/run_main_module.js:18:47

```

**Comment by kanagarajkm** [comment]
`getBucketNotification` returns empty even when the bucket has the config.

**Comment by ebozduman** [comment]
@kanagarajkm 
Addressed bucket notification part of the problem also. 
PTAL again.

**Comment by ebozduman** [comment]
Looking into the build issue

**Comment by kannappanr** [comment]
@kanagarajkm PTAL

**Comment by SamuelHassine** [comment]
@ebozduman @kannappanr We need this PR to be merged and minio-js to be released before tomorrow because this error makes our application crashes and we plan a huge/major release on our side. Do you think that might be possible?

**Review by kanagarajkm — APPROVED** [review]
LGTM

**Review by kannappanr — APPROVED** [review]

**Comment by PacoDu** [comment]
Why does this PR revert this one https://github.com/minio/minio-js/pull/861/files ??

**Comment by ebozduman** [comment]
@PacoDu 

Actually it doesn't revert back the changes. We still use faster xml parser, `fxp`, for parsing and get the benefit.
We had to introduce go back to `xml2js` since we also need another functionality, conversion from json to xml, that does not come with `fxp`. That's why we brought back the `xml2js`.

**Pull request merged**